src/generate/animation/singletemplate.cpp

```diff
@@ -7,2670 +9,3145 @@
-			{
-				singleCondition(condition, format, behaviorFile, multiOption, numline, isGroup, isMaster, optionlist);
-			}
-
-			// OR exist but not AND
-			else
-			{
-				OR = true;
-				singleCondition(condition.substr(0, c_or), format, behaviorFile, multiOption, numline, isGroup, isMaster, optionlist);
-				next = make_shared<condt>(condition.substr(c_or + 1), format, behaviorFile, original, multiOption, numline, isGroup, isMaster, optionlist);
-			}
-		}
-
-		// AND exist but not OR
-		else if (c_or == NOT_FOUND)
-		{
-			singleCondition(condition.substr(0, c_and), format, behaviorFile, multiOption, numline, isGroup, isMaster, optionlist);
-			next = make_shared<condt>(condition.substr(c_and + 1), format, behaviorFile, original, multiOption, numline, isGroup, isMaster, optionlist);
-		}
-
-		// AND & OR exist
-		else
-		{
-			// OR before AND
-			if (c_or < c_and)
-			{
-				OR = true;
-				singleCondition(condition.substr(0, c_or), format, behaviorFile, multiOption, numline, isGroup, isMaster, optionlist);
-				next = make_shared<condt>(condition.substr(c_or + 1), format, behaviorFile, original, multiOption, numline, isGroup, isMaster, optionlist);
-			}
-
-			// AND before OR
-			else
-			{
-				singleCondition(condition.substr(0, c_and), format, behaviorFile, multiOption, numline, isGroup, isMaster, optionlist);
-				next = make_shared<condt>(condition.substr(c_and + 1), format, behaviorFile, original, multiOption, numline, isGroup, isMaster, optionlist);
-			}
-		}
-	}
+    if (condition.find("<") == 0
+        && (condition.find(">") == condition.length() - 1 || condition.find(">*") == condition.length() - 2)
+        && (condition.find("!=") != NOT_FOUND || condition.find("==") != NOT_FOUND))
+    {
+        specialCondition(
+            condition, format, behaviorFile, multiOption, numline, isGroup, isMaster, optionlist);
+    }
+    else
+    {
+        string conditionOrder = condition;
+
+        if (conditionOrder[0] == '!')
+        {
+            isNot          = true;
+            conditionOrder = conditionOrder.substr(1);
+        }
+
+        conditionProcess(conditionOrder, format, behaviorFile, multiOption, numline, isGroup, isMaster);
+    }
 }
 
-multichoice::multichoice(string cond, string format, string behaviorFile, string multiOption, int numline, bool isGroup, bool isMaster, OptionList& optionlist,
-	size_t posA, size_t posB)
+condt::condt(string condition,
+             string format,
+             string behaviorFile,
+             string originalCondition,
+             string multiOption,
+             int numline,
+             bool isGroup,
+             bool isMaster,
+             OptionList& optionlist)
 {
-	if (cond.length() > 0)
-	{
-		condition = make_shared<condt>(cond, format, behaviorFile, cond, multiOption, numline, isGroup, isMaster, optionlist);
-	}
-	else
-	{
-		condition = make_shared<condt>(format, format, behaviorFile, cond, multiOption, numline, isGroup, isMaster, optionlist);
-	}
```

Comment:
why is this part removed?

Reply:
Good find. It took me a few minutes to remember. Actually, it was not removed, it was moved to templateprocessing.cpp
It wouldn't compile without it, I still do not understand. So I did not bother and I moved it